test/models/concerns/key_pair_compute_resource_test.rb

```diff
@@ -0,0 +1,49 @@
+require 'test_helper'
+
+class DummyComputeResource < ComputeResource
+  include KeyPairComputeResource
+
+  def client
+    @client ||= ::Fog::Compute.new(:provider => 'AWS', :aws_access_key_id => '123', :aws_secret_access_key => 'abcd')
```

Comment:
Nitpick, `include Mocha::API` above and then you can simply use `mock('client')` - much lighter than using Fog.